strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/StrongboxBuiltinAuthenticator.java

```diff
@@ -0,0 +1,43 @@
+package org.carlspring.strongbox.authentication.api.impl;
+
+import org.carlspring.strongbox.authentication.api.AuthenticationSupplier;
+import org.carlspring.strongbox.authentication.api.Authenticator;
+import org.carlspring.strongbox.authentication.api.impl.userdetails.StrongboxUserDetailService;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class StrongboxBuiltinAuthenticator
+        implements Authenticator
+{
+
+    @Inject
+    @StrongboxUserDetailService.StrongboxUserDetailServiceQ
```

Comment:
> Also, I don't think we need to prefix everything with Strongbox... Can't we just call it UserDetailService?

StrongboxUserDetailService is an existing implementation, see below for reference:

https://github.com/strongbox/strongbox/blob/master/strongbox-web-core/src/main/java/org/carlspring/strongbox/security/user/StrongboxUserDetailService.java

For the purpose of my task, I moved it to the "strongbox-user-management" module. I needed it there and I don't think StrongboxUserDetailService has anything related with "web" inside it. I also included some adjustments required for my implementation.

So if you mind changing this name, I can do that, but please remember that `StrongboxUserDetailService implements UserDetailsService` so changing StrongboxUserDetailService name to UserDetailsService may be misleading.

> What does the Q stand for...? :)

It's a Qualifier for dependency injection that works in conjunction with @Inject annotation, see https://github.com/fuss86/strongbox/blob/SB-625/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/userdetails/StrongboxUserDetailService.java#L163 Do you want me to change its name ?